packages/theme/src/cli/utilities/ruby.ts

```diff
@@ -0,0 +1,84 @@
+import {path, ui, output} from '@shopify/cli-kit'
+import {exec} from '@shopify/cli-kit/node/system'
+import {
+  bundleExecutable,
+  bundleInstallLocalShopifyCLI,
+  validateRubyEnv,
+  ExecCLI2Options,
+  execCLI2,
+} from '@shopify/cli-kit/node/ruby'
+import {AbortSilentError} from '@shopify/cli-kit/node/error'
+import {fileURLToPath} from 'url'
+
+/**
+ * Execute Theme CLI commands which is an embed simplified version of the CLI2.
+ * Installs a version of RubyCLI as a vendor dependency in a hidden folder in the system.
+ * User must have a valid ruby+bundler environment to run any command.
+ *
+ * @param args - List of argumets to execute. (ex: ['theme', 'pull']).
+ * @param options - Options to customize the execution of cli2.
+ */
+export async function execCLI(args: string[], options: ExecCLI2Options = {}): Promise<void> {
```

Comment:
Instead of creating this new utility and function, that requires changing all the theme commands, wouldn't it be easier to just add the new logic inside execCLI2 (from cli-kit/src/public/node/ruby) and run the required CLI from there (RubyGems or embedded)?

Reply:
Also, it seems you missed to replace execCLI2 in the dev command, which is required to run Theme App extensions. Another reason to move this to cli-kit.